components/ResourceCounts.tsx

```diff
@@ -39,12 +41,42 @@ const ResourceCounts = () => {
       });
   }, [context.countChange]);
 
+  /**
+   * sorts the keys in descending order based on their count, and filters out keys based on the value from the searchbar
+   * @params an object with a string: number key value pair
+   * @returns array of sorted and filtered keys and values
+   */
+  const sortResourceArray = (toSort: { [x: string]: number }): string[] => {
+    if (searchValue) {
+      //filters the array based on searchValue
+      const filteredArray = Object.keys(toSort).filter((el) =>
+        el.toLowerCase().includes(searchValue),
+      );
+      return filteredArray.sort((a, b) => {
+        return toSort[b] - toSort[a];
+      });
+    }
+
+    //sorts keys in descending order based on their value
+    return Object.keys(toSort).sort((a, b) => {
+      return toSort[b] - toSort[a];
+    });
+  };
+
   /**
    * Sorts the resource counts key:value pairs object, then returns them as an array of buttons
    * @returns array of JSX Buttons that are the sorted resources and their counts
    */
-  const getResourceCountsNodes = () => {
-    return sortResourceArray(resources).map((resourceType) => (
+  const GetResourceCountsNodes = () => {
+    let keyCount: string[] = sortResourceArray(resources);
+
+    //filters the resource array when searchValue updates
+    useEffect(() => {
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      keyCount = sortResourceArray(resources);
+    }, [searchValue]);
+
+    const myarray = keyCount.map((resourceType) => (
```

Comment:
setting the variable like this in a useEffect is actually not best practice. I think the solution can be a bit simpler.

You have a function that sorts an array based on some state variable, and you are updating the sorted list with each change of the value of a state variable. Do we even need the useEffect? Can't we just call .map on the sorted array itself? E.g.:

```
sortResourceArray(resources).map((resourceType)...
```

give it a try ;) 
```